editor/editor_help.cpp

```diff
@@ -460,8 +526,9 @@ void EditorHelp::_pop_code_font() {
 	class_desc->pop();
 }
 
-Error EditorHelp::_goto_desc(const String &p_class, int p_vscr) {
-	if (!doc->class_list.has(p_class)) {
+Error EditorHelp::_goto_desc(const String &p_class) {
+	// If class doesn't have docs listed, attempt on-demand docgen
+	if (!doc->class_list.has(p_class) && !_attempt_doc_load(p_class)) {
```

Comment:
So this is an example of `doc` being accessed without checking for the thread, which seems to be problematic already. But now we're also trying to load a script, parse the docs, and then generate editor help. So this is either going to hitch, or going to fail while threads finish.

Perhaps this is not critical at the moment, as it shouldn't be _worse_ than now, but further testing may be needed.

Reply:
For right now I'm leaving the comments so someone with better threading knowledge than me can take a look in the future and marking this as resolved for the purposes of this PR :)